### PR TITLE
Update demo for lvgl 7.9.1

### LINKED ISSUE #19: Update to 7.0.0 Please!
First of all, thanks for this marvelous software and congratulation in your news achievements!

I'm so exited to test out the 7.0.0 version.
But it will be Wonderfull if you could update this simulator!

Thanks in advance.

### PR DESCRIPTION
**Pull request by bjwschaap**
I updated this repo to use the latest available platformio lvgl library version (7.9.1), and rewrote the demo to use the new v7 API.

Fixes #19

### CODE REVIEW

**Comment by bjwschaap** [comment]
Fixes #19

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -57,6 +57,8 @@ jobs:
     runs-on: windows-latest
     steps:
       - name: Activate msys2 & mingw64
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: true
```
Comment (embeddedt) [inline .github/workflows/ci.yml:61]: Ideally the `add-path` commands need to be fixed, not just bypassed. At first glance an environment file should work for what we're doing here.

https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

**Review by embeddedt — COMMENTED** [review]
@bjwschaap Thank you for these updates! I think we would want to replace this demo with the more modern equivalent which is [`lv_demo_widgets`](https://lvgl.github.io/lv_examples/lv_demo_widgets/index.html), but let's wait for @puzrin's review.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (puzrin) [reply .github/workflows/ci.yml:61]: @embeddedt see `ci*` branches in https://github.com/puzrin/dispenser. I failed to fix that in better way. If anyone will be able to do that - cool (but not mandatory)

**Comment by puzrin** [comment]
@bjwschaap thank you very much for your work!

If anyone wish to do more cleanup, here are 2 directions:

- Drop `/src/lv_conf.h` and move `-D` overrides to `platform.ini`.
- Remove use of `add-path` from CI.

**Comment by bjwschaap** [comment]
@puzrin No problem, happy to help!

Could you please elaborate on what you mean with: Drop /src/lv_conf.h and move -D overrides to platform.ini ?
Because to my understanding using `lv_conf.h` is a standard lvgl way of configuring. Switching to an alternative way of configuring lvgl might be confusing for people new to this project/codebase. Maybe create a new/separate issue for this?

**Comment by puzrin** [comment]
> Could you please elaborate on what you mean with: Drop /src/lv_conf.h and move -D overrides to platform.ini ?
> Because to my understanding using `lv_conf.h` is a standard lvgl way of configuring. Switching to an alternative way of configuring lvgl might be confusing for people new to this project/codebase.

At some moment (in v7), lvgl added built-in config https://github.com/lvgl/lvgl/blob/master/src/lv_conf_internal.h

This was not possible with v6 (when this repo was created). But now you can just place `-Dxxxx` into `platform.ini`. I think, this would be useful. Because this project shows "how to use platformio (not how to use lvgl).

> Maybe create a new/separate issue for this?

If that helps - do as you wish. I'm not actively participating in this repo. It was created with v6, and now i'm waiting v8. Usually, i experiment with my own projects, and then copy useful things here.

**Comment by puzrin** [comment]
I solved most of issues. Now demo is up to date with PIO v5+ featues. See changelog.

The only thing left is to change OSX compiler. If anyone knows how to do so - please take a look.

**Comment by bjwschaap** [comment]
Awesome @puzrin !!! Thank you for sharing this repo!

**Pull request merged**